src/restclient/backup/backupresult.ts

```diff
@@ -1,5 +1,57 @@
-export enum BackupResult {
+export enum BackupResultState {
     STATE_STOP = "stop",
     STATE_BACKUP = "backup",
     STATE_RESTORE = "restore"
-}
\ No newline at end of file
+}
+
+export enum BackupResultResult {
+    RESULT_SUCCESS = "success",
+    RESULT_FAILED = "failed",
+    RESULT_PROCESS = "process"
+}
+
+export class BackupResult {
+    private _state: string;
+    private _result: string;
+    private _message: string;
+
+    get state(): BackupResultState {
+        if (this._state === 'stop') {
+            return BackupResultState.STATE_STOP;
+        } else if (this._state === 'backup') {
+            return BackupResultState.STATE_BACKUP;
+        } else if (this._state === 'restore') {
+            return BackupResultState.STATE_RESTORE;
+        } else {
+            throw Error('Unknown state.');
+        }
+    }
+
+    set state(value: string) {
+        this._state = value;
+    }
+
+    get result(): BackupResultResult {
+        if (this._result === 'success') {
+            return BackupResultResult.RESULT_SUCCESS;
+        } else if (this._state === 'failed') {
+            return BackupResultResult.RESULT_FAILED;
+        } else if (this._state === 'process') {
+            return BackupResultResult.RESULT_PROCESS;
+        } else {
+            throw Error('Unknown result.');
+        }
+    }
```

Comment:
Validation errors should occur on setters, not on getters. It's a bit late to get an error there. And BackupResult should be a plain object, which means that it simply transport data. If we want to validate its content, then we need a constructor or a factory method with validation. This would allow the error to occur at the right place.

Reply:
In fact, this is an internal error. If the error occurs, which means `hive node` server has a bug.

Reply:
Yes, I understand that, that's why I was talking about a validation in a constructor or a factory to prevent adding too much logic code in a simple plain object. The class is still a bit heavy for a plain object, but I understand that you want the implementer to use the enums to validate the content of the result object. I'm fine with that.